packages/cli/src/util/aws/checkImageExists.ts

```diff
@@ -0,0 +1,37 @@
+import { ListImagesCommand } from '@aws-sdk/client-ecr';
+import { getECRClient } from './clients/getECRClient';
+import type { PanfactumContext } from '@/util/context/context';
+
+/**
+ * Check if an image with the specified tag exists in an ECR repository
+ * 
+ * @param accountId - AWS account ID that owns the ECR registry
+ * @param repoName - ECR repository name
+ * @param tag - Image tag to search for
+ * @param region - AWS region where the ECR repository is located
+ * @param context - Panfactum context for logging and AWS credentials
+ * @returns Promise<boolean> - true if image exists, false otherwise
+ */
+export async function checkImageExists(
+  accountId: string,
+  repoName: string,
+  tag: string,
+  region: string,
+  context: PanfactumContext
+): Promise<boolean> {
+  try {
+    const client = await getECRClient({ context, region });
+
+    const result = await client.send(new ListImagesCommand({
+      registryId: accountId,
+      repositoryName: repoName
+    }));
+
+    // Check if any image has the specified tag
+    return result.imageIds?.some(image => image.imageTag === tag) ?? false;
+  } catch (error) {
+    // Log the error but don't throw - let the caller handle retry logic
+    context.logger.warn(`Error checking for image: ${error instanceof Error ? error.message : 'Unknown error'}`);
```

Comment:
No. Don't print this out??? The error should be thrown.

Returning `false` makes absolutely zero sense. You don't know whether the image exists if there is an error so you cannot return either true or false... you can only throw an error.